Added all upgrade docs (#4925)
umbraco/UmbracoDocs: umbraco-cloud/upgrades/version-specific-upgrades/migrate-from-umbraco-7-to-8.md

---
description: This article will provide detailed steps on how to migrate a Umbraco 7 Cloud project to Umbraco 8.
---

# Migrate from Umbraco 7 to Umbraco 8 on Umbraco Cloud

> Taking your Umbraco CMS project from Umbraco 7 to 8 is called a migration as it requires that the data is migrated in the process. This article covers each step involved in this process.

Reviewer: ```suggestion
Taking your Umbraco CMS project from Umbraco 7 to 8 is called migration as it requires that the data is migrated in the process. This article covers each step involved in this process.
```

> Read the [general article about Content migration](https://docs.umbraco.com/umbraco-cms/fundamentals/setup/upgrading/version-specific/migrate-content-to-umbraco-8.md#what-are-the-limitations) to learn more about limitations and other things related to migrating your Umbraco site from 7 to 8.

Reviewer: Link is not working (Broken link)

Author: That's because it's linking to an article that will be added with this PR ;)

Reviewer: ah ok.. got it.. 😀

## Video tutorial

<iframe width="800" height="450" title="Migrate an Umbraco Cloud project from 7 to 8: Introduction" src="https://www.youtube.com/embed/wNIDdgdAt8s?rel=0" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe>

> You can find the full playlist here: [Migrate an Umbraco Cloud project from 7 to 8](https://www.youtube.com/playlist?list=PLgX62vUaGZsGceCXveMkprlDV5a3K94db)

Reviewer: The videos in this playlist are not in order.

Author: This has now been sorted 💪

## Prerequisites

* A Umbraco 7 Cloud project running **the latest version of Umbraco 7**.
* Make sure Umbraco Forms data is not handled as content.
  * See [Umbraco Forms on Cloud](../../deployment/umbraco-forms-on-cloud.md#how-forms-are-handled-on-umbraco-cloud) for more details on how to check the setting.
* A clean Cloud project running the latest version of Umbraco 8 with **at least 2 environments**.

{% hint style="info" %}
We strongly recommend having at least 2 environments on the Umbraco 8 project.

Should something fail during the migration, the Development environment can always be removed and re-added in order to start over on a clean slate.
{% endhint %}

## Step 1: Content migration

1. Clone down the Umbraco 7 Cloud project.
2. Run the project locally and **restore** Content and Media.

> 3. Clone down the Development environment from the Umbraco 8 Cloud project.

Reviewer: I think it would be helpful to add a tip or note linking to this article: https://github.com/umbraco/UmbracoDocs/blob/legacy-docs/Umbraco-Cloud/Set-Up/Working-Locally/Working-With-Visual-Studio/index.md (in case, someone doesn't know how to generate a .csproj file).

Also, I can find the updated version of this article on the docs site, not the v7-v8 version.

Author: Using the UaaS.cmd tool is an outdated way of working with Cloud, so I do not think we should link to it.
Also, it's not something that we recommend as the tool isn't being maintained by HQ anymore.

We could add a link to the tool to this article instead, perhaps 🤔 
With a few notes on how to use it, and with the message that this is only a tool to be used for this purpose...

Reviewer: Yup, some tips/info around creating a .csproj will be helpful considering the next step to install a package requires one. It may or may not be useful but it can be helpful to just put it out there 😀

Author: I added a note between bullet 3 and 4.
Let me know what you think!

Reviewer: Looks good to me 🙌

4. Install the [ProWorks Umbraco 8 Migrations](https://www.nuget.org/packages/ProWorks.Umbraco8.Migrations) community package on the cloned Umbraco 8 site.
5. Copy `~/App_Data/Umbraco.sdf` or `~/App_Data/Umbraco.mdf` from the cloned Umbraco 7 project.
6. Paste the file into `~/App_Data` on the clone of the Umbraco 8 project.
7. Open `web.config` from the Umbraco 8 project.
8. Locate the `Umbraco.Core.ConfigurationStatus` key.
9. Replace the value with the version your Umbraco 7 project is running.
10. Run the Umbraco 8 project locally
11. Authorize the migration - Cloud credentials are used for this.

> ![Authorize upgrade](../images/upgrade-to-8_1.png)

Reviewer: Do you want to change the image to show the latest version of 8 in the message? You can use this for reference.
![image](https://user-images.githubusercontent.com/82437098/222170318-ffade9da-2605-442f-aa8e-223d2bc68cef.png)

Author: That's a great idea! I've grabbed the image - will upload it to the article.

12. Click **Continue** to start the migration.
13. Log in to the backoffice and verify that everything is there once the migration is complete.

{% hint style="info" %}
If your login does not work, try the following approach:

> 1, Copy the `UsersMembershipProvider` attributes from your Umbraco 7 `web.config`, to the Umbraco 8 `web.config`.

Reviewer: ```suggestion
1. Copy the `UsersMembershipProvider` attributes from your Umbraco 7 `web.config` file to the Umbraco 8 `web.config` file.
```

> 2. Try to login again.

Reviewer: The list items are displayed as a sentence on GitBook instead of list. Not sure if it's because it's in a Note.

Author: Hm. Yeah! It might be. I don't think this needs a list anyways, so I'll rewrite it a bit.

Below is an example of how the attribute can look:

```xml
<add name="UsersMembershipProvider" 
     type="Umbraco.Web.Security.Providers.UsersMembershipProvider, Umbraco" 
     minRequiredNonalphanumericCharacters="0" 
     minRequiredPasswordLength="8" 
     useLegacyEncoding="true" 
     enablePasswordRetrieval="false" 
     enablePasswordReset="true" 
     requiresQuestionAndAnswer="false" 
     passwordFormat="Hashed" />
```

{% endhint %}

{% hint style="info" %}
Please be aware that this is **only a content migration**.

The database will be migrated, but updating view files, custom code, and implementation is a manual process.

See [Step 3](#step-3-setup-custom-code-for-umbraco-8) of this guide, for more detail on this.
{% endhint %}

## Step 2: Files migration

> Before moving on with the next step, make sure that the Umbraco 8 project is no longer running.

Reviewer: ```suggestion
Before moving on to the next step, make sure that the Umbraco 8 project is no longer running.
```

The following files/folders need to be copied into the Umbraco 8 project:

* `~/Views` - do **not** overwrite the default Macro and Partial View Macro files, unless changes have been made to these.
* `~/Media`
* Any files/folders related to Stylesheets and JavaScripts.
* Any custom files/folders the Umbraco 7 project uses, that aren't in the `~/Config` or `~/bin`.
* `~/App_Data/UmbracoForms` - in the case Umbraco Forms was used on the Umbraco 7 site.

> Merge the configuration files carefully to ensures any custom settings are migrated while none of the default configurations for Umbraco 8 is overwritten.

Reviewer: ```suggestion
Merge the configuration files carefully to ensure any custom settings are migrated while none of the default configurations for Umbraco 8 is overwritten.
```

### Generating UDA files

1. Run the Umbraco 8 project locally
    * It **will** give you an error on the frontend as none of the Template files have been updated yet.

> 2. Open the commandline tool in the `~/data` folder on the Umbraco 8 project.

Reviewer: ```suggestion
2. Open the command line tool in the `~/data` folder on the Umbraco 8 project.
```

* `deploy-complete`
  * Everything checks out: Move on to the next step

## Step 3: Setup custom code for Umbraco 8

Umbraco 8 is different from Umbraco 7 in many ways. This means that in this step, all custom code, controllers, and models need to be reviewed and rewritten for Umbraco 8.

### Example of changes that need to be made

One of the changes made, is how published content is rendered through Template files. Due to this, it will be necessary to update **all** Template files (`.cshtml`) to reflect these changes.

Read more about these changes in the [IPublishedContent section of the Documentation](https://docs.umbraco.com/umbraco-cms/reference/querying/ipublishedcontent).

* Template files need to inherit from `Umbraco.Web.Mvc.UmbracoViewPage<ContentModels.HomePage>` instead of `Umbraco.Web.Mvc.UmbracoTemplatePage<ContentModels.HomePage>`.
* Template files need to use `ContentModels = Umbraco.Web.PublishedModels` instead of `ContentModels = Umbraco.Web.PublishedContentModels`.
* `@Model.Value("propertyAlias")` replaces `@Umbraco.Field("propertyAlias")`.
* `@Model.PropertyAlias` replaces `@Model.Content.PropertyAlias`.

> 

Reviewer: I also had to make this change in the template to view the frontend:
`Model.Value<IPublishedContent>("propertyAlias");` replaces `Model.Content.GetPropertyValue<IPublishedContent>("propertyAlias");`

Author: Ah, sure! I didn't test update the rendering, as that's not really something we document anyways, so must have missed this one.

Does it replace the third bullet, or is just a new bullet entirely? 🤔

> Depending on the size of the project and amount of custom code and implementations, this step is going to require a lot of work.

Reviewer: ```suggestion
Depending on the size of the project and the amount of custom code and implementations, this step is going to require a lot of work.
```

## Step 4: Deploy and test on Umbraco Cloud

Once the Umbraco 8 project runs without errors on the local setup, the next step is to deploy and test on the Cloud Development environment.

1. Push the migration and changes to the Umbraco Cloud Development environment

{% hint style="info" %}
The deployment might take a bit longer than normal.

To track the process, keep an eye on the deploy markers in `site/wwwroot/data` using KUDU.
{% endhint %}

2. Progress through the steps based on the deployment result:
    * `deploy-failed`: Run `echo > deploy-clearsignatures` followed by `echo > deploy` to clear up the error.
    * `deploy-complete`: The Development environment has been upgraded.
3. Transfer Content and Media from the local clone to the Development environment.
4. Test **everything** on the Development environment.
5. Deploy to the Live environment.

## Step 5: Going live

Once the migration is complete, and the Live environment is running without errors, the site is almost ready for launch.

1. Setup rewrites on the Umbraco 8 site.
2. Assign hostnames to the project.
    * Hostnames are unique, and can only be added to one Cloud project at a time.

## Related information

> * [Content Migration for Umbraco CMS - 7 to 8](https://docs.umbraco.com/umbraco-cms/fundamentals/setup/upgrading/version-specific/migrate-content-to-umbraco-8.md)

Reviewer: Link doesn't work (Broken Link)

Author: The article linked to, will be added with this PR. That's why the link doesn't work at the moment.

Reviewer: Got it.. 💪

You can find the full playlist here: [Migrate an Umbraco Cloud project from 7 to 8](https://www.youtube.com/playlist?list=PLgX62vUaGZsGceCXveMkprlDV5a3K94db)

## Prerequisites

* A Umbraco 7 Cloud project running **the latest version of Umbraco 7**.
* Make sure Umbraco Forms data is not handled as content.
  * See [Umbraco Forms on Cloud](../../deployment/umbraco-forms-on-cloud.md#how-forms-are-handled-on-umbraco-cloud) for more details on how to check the setting.
* A clean Cloud project running the latest version of Umbraco 8 with **at least 2 environments**.

{% hint style="info" %}
We strongly recommend having at least 2 environments on the Umbraco 8 project.

Should something fail during the migration, the Development environment can always be removed and re-added in order to start over on a clean slate.
{% endhint %}

## Step 1: Content migration

1. Clone down the Umbraco 7 Cloud project.
2. Run the project locally and **restore** Content and Media.
3. Clone down the Development environment from the Umbraco 8 Cloud project.

{% hint style="info" %}

> We recommend setting up the Umbraco 8 Cloud portal locally Visual Studio.

Reviewer: ```suggestion
We recommend setting up the Umbraco 8 Cloud portal locally in Visual Studio.
```

This can be done after cloning down the Cloud environment or by using the [UaaS cloning tool](https://umbra.co/uaas-cmd).

> To use the cloning tool, place it and run it in the local directory you want to clone to Cloud project into.

Reviewer: ```suggestion
To use the cloning tool, place it and run it in the local directory you want to clone the Cloud project into.
```
